error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/CanonicalConstantFieldNameTest.java

```diff
@@ -0,0 +1,156 @@
+package tech.picnic.errorprone.bugpatterns;
+
+import com.google.errorprone.BugCheckerRefactoringTestHelper;
+import com.google.errorprone.BugCheckerRefactoringTestHelper.TestMode;
+import com.google.errorprone.CompilationTestHelper;
+import org.junit.jupiter.api.Test;
+
+final class CanonicalConstantFieldNameTest {
+  @Test
+  void identification() {
+    CompilationTestHelper.newInstance(CanonicalConstantFieldName.class, getClass())
+        .addSourceLines(
+            "A.java",
```

Comment:
The reporting here is a bit weird. It's reported on a class level... and only one error per time while we are creating a builder with multiple fixes... I'm not sure what I'm doing wrong or missing here.
 
It should be:
```
"  // BUG: Diagnostic contains:",
"  private static final int foo = 1;",
"  // BUG: Diagnostic contains:",
"  private static final int foobar = 1;",
"  // BUG: Diagnostic contains:",
"  private static final int foobarbaz = 1;",
```

Later we should drop the `foobar` and `foobarbaz` but now it's here to show what is wrong and we should figure out.

Reply:
So it's reported on the class file... It could be that this is a bug in the SuggestedFixes?

Reply:
> It could be that this is a bug in the SuggestedFixes?

I don't think it is a bug, since we explicitly passing the class tree to the `describeMatch` method. If we want to improve on this, I think we can use a TreeScanner. I will look into this 🙏

Reply:
Pushed something 👍

Reply:
Added a few more tests :).